src/components/PostsList.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Search } from './Search';
+import debounce from '../helpers/debounce';
+import { setQuery } from '../redux/store/query';
+import { useDispatch } from 'react-redux';
+import { Post } from './Post';
+
+export const PostsList = ({ posts }: PostsListProps) => {
+  const [filteredQuery, setFilteredQuery] = useState('');
+  const dispatch = useDispatch();
+
+  const visiblePosts = useMemo(() => {
+    const filteredPosts = [...posts].filter(post => {
+      const title = post.title.toLowerCase();
+      const body = post.body.toLowerCase();
+
+      return (title + body).includes(filteredQuery.toLowerCase());
+    });
+
+    return filteredPosts;
+  }, [posts, filteredQuery]);
+
+  const setFilteredQueryWithDebounce = useCallback(
+    debounce(setFilteredQuery, 1000),
+    [],
+  );
+
+  const handleSearch = useCallback((e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    dispatch(setQuery(e.target.value));
```

Comment:
there is no need to store query for this component in the main state, create local state for such data

Reply:
✅